### PR TITLE
Adding all Esp8266 unit tests and fixing up implementation

### PR DESCRIPTION
**Pull request by PricelessAntonio**

### CODE REVIEW

File: library/L2_HAL/communication/test/esp8266_test.cpp [hunk 1]
```diff
@@ -19,7 +19,46 @@ TEST_CASE("Esp8266 Test", "[esp8266]")
...
+          data[i] = kConfirmationResponse[i];
+        }
+        count++;
+      }
+      // The second uart read is used to flush the serial buffer
+      else
+      {
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == 1);
+      }
```
Comment (kammce) [inline library/L2_HAL/communication/test/esp8266_test.cpp]: Can you add an issue to convert this into a template test, or something more generic. You echo this multiple times throughout your test code.

File: library/L2_HAL/communication/test/esp8266_test.cpp [hunk 2]
```diff
@@ -60,15 +161,167 @@ TEST_CASE("Esp8266 Test", "[esp8266]")
...
+    Verify(
+        ConstOverloadedMethod(mock_uart, Write, void(const uint8_t *, size_t))
+            .Using(Esp8266::kGetNetworkConnectionInfoCommand,
+              (sizeof(Esp8266::kGetNetworkConnectionInfoCommand)-1)));
   }
+
   SECTION("Connect to Access Point")
   {
+    static constexpr uint8_t kExpectedWrite[] =
+        "AT+CWJAP=\"ssid_test\",\"password_test\"\r\n";
```
Comment (kammce) [inline library/L2_HAL/communication/test/esp8266_test.cpp]:
```suggestion
        R"(AT+CWJAP="ssid_test","password_test"\r\n)";
```
You can use a raw string here and get rid of the escape characters.

Comment (PricelessAntonio) [reply library/L2_HAL/communication/test/esp8266_test.cpp]: For this one, I do need the escaped characters for \r and \n though.

File: library/L2_HAL/communication/test/esp8266_test.cpp (hunk 1 above)
Comment (PricelessAntonio) [reply library/L2_HAL/communication/test/esp8266_test.cpp]: Alright

**Comment by kammce** [comment]
Closing as this pull request is stale.

**Pull request closed without merging**